PurchasesCoreSwift/Purchasing/ReceiptFetcher.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  ReceiptFetcher.swift
+//  Purchases
+//
+//  Created by Javier de Martín Gil on 8/7/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+// TODO: Make internal after migration to Swift is complete
+@objc(RCReceiptFetcher) public class ReceiptFetcher: NSObject {
+    private let requestFetcher: StoreKitRequestFetcher
+    private let receiptBundle: Bundle
+
+    @objc public convenience init(requestFetcher: StoreKitRequestFetcher) {
+        self.init(requestFetcher: requestFetcher, bundle: .main)
+    }
+
+    init(requestFetcher: StoreKitRequestFetcher, bundle: Bundle) {
+        self.requestFetcher = requestFetcher
+        self.receiptBundle = bundle
+    }
+
+    @objc public func receiptData(refreshPolicy: ReceiptRefreshPolicy,
+                                  completion: @escaping ((Data?) -> Void)) {
+        if refreshPolicy == .always {
+            Logger.debug(String(format: Strings.receipt.force_refreshing_receipt))
+            self.refreshReceipt(completion)
+            return
+        }
+
+        let receiptData = receiptData()
+        let isReceiptEmpty = receiptData?.isEmpty ?? true
+
+        if isReceiptEmpty && refreshPolicy == .onlyIfEmpty {
+            Logger.debug(Strings.receipt.refreshing_empty_receipt)
+            self.refreshReceipt(completion)
+        } else {
+            completion(receiptData)
+        }
+    }
+
+}
+
+private extension ReceiptFetcher {
+
+    func receiptData() -> Data? {
```

Comment:
this method used to be public, but now it's only used within this class

Reply:
are we keeping a list somewhere of changes to the public interface?

Reply:
I think I should have been more clear: 
the method was originally `internal`, then it became temporarily `public` for the migration, but what I _meant_ is that it's part of the interface for the class. 
As for list of changes to the public interface, yeah, we have it listed in a google doc (DM me for the link if you need it). This isn't listed in it since the class used to be internal and will be made internal again once the migration is done